src/db/spanner/models.rs

```diff
@@ -1250,41 +1233,40 @@ impl SpannerDb {
             sqlparams.insert("newer".to_string(), as_value(newer.as_rfc3339()?));
             sqltypes.insert("newer".to_string(), as_type(TypeCode::TIMESTAMP));
         }
+
+        // if the offset has not yet been applied (we set it to None if we have), then apply it using the
```

Comment:
In the future we may want to remove the ability to even do limit/offset queries. It would greatly simplify the code.

Reply:
probably. I can still kinda/sorta see the value of a limit, but agree that not having an offset would be good.

The only problem I had when I first did that was that while we have a "newer" and "older", they're more strict than we want a given offset to be. (e.g. you should never have a records = to a newer or older, while you absolutely will have one when doing an offset.)